src/lib/logical_query_plan/mock_node.cpp

```diff
@@ -28,14 +28,23 @@ LQPColumnReference MockNode::get_column(const std::string& column_name) const {
 const MockNode::ColumnDefinitions& MockNode::column_definitions() const { return _column_definitions; }
 
 const std::vector<std::shared_ptr<AbstractExpression>>& MockNode::column_expressions() const {
+  // Need to initialize the expressions lazily because they will have a weak_ptr to this node and we can't obtain that
+  // in the constructor
   if (!_column_expressions) {
-    _column_expressions.emplace();
+    _column_expressions.emplace(_column_definitions.size() - _pruned_column_ids.size());
 
-    const auto column_count = _column_definitions.size();
+    auto pruned_column_ids_iter = _pruned_column_ids.begin();
 
-    for (auto column_id = ColumnID{0}; column_id < column_count; ++column_id) {
-      const auto column_reference = LQPColumnReference(shared_from_this(), column_id);
-      _column_expressions->emplace_back(std::make_shared<LQPColumnExpression>(column_reference));
+    for (auto stored_column_id = ColumnID{0}, output_column_id = ColumnID{0};
```

Comment:
Philosophical, I guess. We'd then have a variable on scope higher than it would need to be, which is something we'd usually try to avoid. But I removed it from the loop initialization.